pygmt/clib/loading.py

```diff
@@ -151,5 +151,10 @@ def check_libgmt(libgmt):
     functions = ["Create_Session", "Get_Enum", "Call_Module", "Destroy_Session"]
     for func in functions:
         if not hasattr(libgmt, "GMT_" + func):
-            msg = f"Error loading libgmt. Couldn't access function GMT_{func}."
+            # pylint: disable=protected-access
+            msg = (
+                f"Error loading '{libgmt._name}'. "
+                f"Couldn't access function GMT_{func}. "
+                "Maybe loading an old version of the GMT shared library."
```

Comment:
This error message does look better, more explicit but I think we can do better and make it more constructive (see https://www.nngroup.com/articles/error-message-guidelines/). How about:

```suggestion
                f"Error loading '{libgmt._name}'. Couldn't access function GMT_{func}. "
                "Ensure that you have installed an up to date GMT version 6 library. "
                "Use `pygmt.show_versions()` to check if an old version of the GMT "
                "shared library (e.g. GMT 5) was accidentally loaded."
```

Reply:
Good suggestion.

>                 "Ensure that you have installed an up to date GMT version 6 library. "

This one sounds good. 

>                 "Use `pygmt.show_versions()` to check if an old version of the GMT "
>                 "shared library (e.g. GMT 5) was accidentally loaded."

Users can not run `pygmt.show_versions()` if incorrect library was loaded.

Reply:
Ah right, the error is raised on `import pygmt`. Hold on, lemme rephrase it.

Reply:
```suggestion
                f"Error loading '{libgmt._name}'. Couldn't access function GMT_{func}. "
                "Ensure that you have installed an up to date GMT version 6 library. "
                "Please check that an old version of the GMT shared library "
                "(e.g. GMT 5) was not accidentally loaded."
```

Reply:
>               "Please check that an old version of the GMT shared library "
>               "(e.g. GMT 5) was not accidentally loaded."

Users may not know how to check the library version. Even they know, they may still not know how to load the correct one.

What about:

>                f"Error loading '{libgmt._name}'. Couldn't access function GMT_{func}. "
>                "Ensure that you have installed an up-to-date GMT version 6 library. "
>                "Please set the environment variable 'GMT_LIBRARY_PATH' to the directory "
>                "of the GMT 6 library."

Reply:
Yep, looks ok.

Reply:
Done in b004fc95099172217295bbd41b3089ba9ea3a993.